Feat: PosterGallery 컴포넌트 구현

thsk3 opened the pull request:
## 📝 Summary
<!-- PR에 대한 간략한 설명을 적어주세요 -->
공연 포스터를 그리드로 나타낸 컴포넌트입니다.

## ✨ Changes
<!-- 변경된 내용들을 상세히 나열해 주세요 -->
- PosterGallery 컴포넌트 추가
- PosterWithDesc 컴포넌트에 공연장 정보 추가

## 🖼️ Screenshots
<!-- 필요한 경우 스크린샷을 첨부해 주세요 -->
<img width="538" alt="image" src="https://github.com/user-attachments/assets/6cda46cf-cad4-4cc3-9d0b-7787d9e4a095">
<img width="384" alt="image" src="https://github.com/user-attachments/assets/82487bcb-b89c-477d-8ffd-19b71b28aace">

## ✅ PR Checklist
<!-- 코드 리뷰 시 확인해야 할 사항들 -->
- [x] 하나의 목적을 가진 PR입니다.
- [x] 코딩 컨벤션과 스타일 가이드를 준수합니다. [📌Conventions](https://github.com/prgrms-fe-devcourse/NFE1-2-LocalStage/wiki/%F0%9F%93%8CConventions)
- [x] 불필요한 코드 중복이 없습니다.
- [x] 컴포넌트의 책임이 단일합니다.
- [ ] 리액트 훅을 올바르게 사용했습니다.
- [ ] 민감한 정보를 포함하지 않았습니다.
- [x] 불필요한 콘솔 로그나 주석을 포함하지 않았습니다.
- [ ] 컴포넌트 key값에 고유한 값을 할당했습니다.

## 🔗 References
<!-- 관련된 이슈, PR, 링크 등을 첨부해 주세요 -->
- Issue: https://github.com/prgrms-fe-devcourse/NFE1-2-LocalStage/issues/64

## 💬 Comments
<!-- 추가적인 커멘트가 있다면 작성해 주세요 -->

juriurj committed:
Feat: PosterGallery 컴포넌트 구현

juriurj committed:
Feat: PosterWithDesc 컴포넌트 공연장 추가

shlee9999 commented on src/components/PosterGallery/index.tsx:
```diff
@@ -0,0 +1,30 @@
+            place={perform.fcltynm}
+            date={perform.prfpdfrom + '-' + perform.prfpdto}
+            width="100%"
+          ></PosterWithDesc>
```
`<PosterWithDesc /> 형태로 바꾸어 주세요`

shlee9999 commented on src/components/PosterGallery/index.tsx:
```diff
@@ -0,0 +1,30 @@
+export const PosterGallery = ({ pList, width = '600px', gap = 10 }: PosterGalleryProps) => {
+  return (
+    <S.PosterGallery width={width} gap={gap}>
+      {pList.map((perform, index) =>
```
slice로 7개만 받아오게 해주세요!

juriurj committed:
Feat: width, height 숫자 추출 함수 구현
- width, height가 문자열일 때 수학 계산을 위해 숫자만 추출하는 함수를 구현함

juriurj committed:
Refactor: Poster 컴포넌트 props 수정
- rank 속성 옵셔널 값으로 변경

juriurj committed:
Docs: PosterGallery 컴포넌트 pList타입 정리

juriurj committed:
Refactor: Poster 컴포넌트 props 타입 변경
- width만 입력받도록 함
- 입력한 형태(px, % 등)에 따라 height의 형태도 자동으로 변하게 함

juriurj committed:
Refactor: Poster 관련 컴포넌트 정리
- children이 없는 컴포넌트 닫는 방식 변경 (<어쩌구 />로)
- postergallery 컴포넌트 pList 7개로 잘라서 포스터 보여주도록함

shlee9999 commented on src/components/PosterGallery/index.tsx:
```diff
@@ -4,27 +4,32 @@ import { Poster } from '@/components/Poster';
+              width="100%"
+            />
+          ) : (
+            <Poster key={index} src={perform.poster}></Poster>
```
이 부분도 <Poster /> 형태로 바꾸어 주세요!

thsk3 replied:
넵 수정했습니다!

juriurj committed:
Refactor: PosterGallery 컴포넌트 닫는 방법 수정
